src/clima_sensor_us.rs

```diff
@@ -0,0 +1,523 @@
+	("64cb7d79d588b90007d7402e", REG_SUN_ELEVATION, 10.0, 's'),
+	("64cb7d79d588b90007d7402f", REG_SUN_AZIMUTH, 10.0, 's'),
+	("64cb7d79d588b90007d74030", REG_HEIGHT_ABOVE_SEA, 1.0, 's'),
+	("64cb7d79d588b90007d74031", REG_SENSOR_SUPPLY, 10.0, 'u'),
+	("64cb7dfbd588b90007d782fc", REG_MAX_WIND_SPEED, 10.0, 'u'),
+	("64cb7dfbd588b90007d782fd", REG_WIND_DIREC, 10.0, 'u'),
+	("64cb7dfbd588b90007d782fe", REG_ABS_HUMIDITY, 100.0, 'u'),
+	(
+		"64cb7dfbd588b90007d782ff",
+		REG_REL_HUMIDITY_UNCORRECTED,
+		10.0,
+		'u',
+	),
+	(
+		"64cb7eb2d588b90007d7dd96",
+		REG_MAGNETIC_COMPASS_DIFF_ANGLE,
+		10.0,
+		'u',
+	),
+	("64cb7eb2d588b90007d7dd97", REG_BRIGHTNESS_VEC_SUM, 1.0, 'u'),
+	("64cb7eb2d588b90007d7dd98", REG_WINDCHILL_TEMP, 10.0, 's'),
+	("64cb7eb2d588b90007d7dd99", REG_HEAT_INDEX_TEMP, 10.0, 's'),
+	(
+		"64cb7eb2d588b90007d7dd9a",
+		REG_ABS_PRECIPITATION_AMOUNT,
+		1000.0,
+		'u',
+	),
+	("64cb7eb2d588b90007d7dd9b", REG_GLOBAL_RADIATION, 10.0, 's'),
+	(
+		"64cb7eb2d588b90007d7dd9c",
+		REG_PITCH_MAGNETIC_COMPASS_NS,
+		10.0,
+		's',
+	),
+	(
+		"64cb7eb2d588b90007d7dd9d",
+		REG_ROLL_MAGNETIC_COMPASS_EW,
+		10.0,
+		's',
+	),
+];
+
+/// These functions create a single number out of a vector.
+/// The first entry in the vector are the most significant bytes and the second entry are the least significant bytes.
+/// For the unsigned function the input `vec` should be already in two complement, so that the function works right.
+fn conv_vec_to_value_s(vec: Vec<u16>) -> Result<i32, ()> {
+	let usign_val: u32 = (vec[0] as u32) << 16 | (vec[1] as u32);
+	if usign_val == ERROR_CODE_S32 {
+		Err(())
+	} else {
+		Ok(usign_val as i32)
+	}
+}
+
+fn conv_vec_to_value_u(vec: Vec<u16>) -> Result<u32, ()> {
+	let usign_val = (vec[0] as u32) << 16 | (vec[1] as u32);
+	if usign_val == ERROR_CODE_U32 {
+		Err(())
+	} else {
+		Ok(usign_val)
+	}
+}
+
+#[derive(Clone, Copy, PartialEq)]
+pub enum TempWarningStateChange {
+	None,
+	ChangeToCloseWindow,
+	ChangeToWarningTempNoWind,
+	ChangeToWarningTemp,
+	ChangeToRemoveWarning,
+}
+
+#[derive(Clone, Copy, PartialEq)]
+pub enum TempWarning {
+	None,
+	RemoveWarning,
+	CloseWindow,
+	WarningTempNoWind,
+	WarningTemp,
+}
+
+pub struct ClimaSensorUS {
+	ctx: Option<Modbus>,
+	opensensebox_id: String,
+	opensense_access_token: String,
+	warning_active: TempWarning,
+	opensensemap_counter: u32,
+}
+
+impl ClimaSensorUS {
+	pub fn new(config: &mut Config) -> Self {
+		let mut s = Self {
+			ctx: None,
+			opensensebox_id: config.get::<String>("opensensemap/box_id"),
+			opensense_access_token: config.get::<String>("opensensemap/access_token"),
+			warning_active: TempWarning::None,
+			opensensemap_counter: 0,
+		};
+		if config.get_bool("weatherstation/enable") {
+			s.init();
+		}
+		s
+	}
+
+	fn init(&mut self) {
+		self.ctx = Some(
+			Modbus::new_rtu(DEVICE, BAUDRATE, PARITY, DATA_BITS, STOP_BITS)
+				.expect("Error accured while creating new RTU Object"),
```

Comment:
Typo is several times:

```suggestion
				.expect("Error occurred while creating new RTU Object"),
```